examples/pennylane_universal_classifier.py

```diff
@@ -0,0 +1,491 @@
+r"""
+.. _universal_classifier:
+
+Universal Quantum Classifier 
+=============================
+Data-reuploading circuits (Perez-Salinas et al. (2019))
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+**Shahnawaz Ahmed (shahnawaz.ahmed95@gmail.com)**
+
+A single-qubit quantum circuit which can implement arbitrary unitary
+operations can be used as a universal classifier much like a single
+hidden-layered Neural Network. As surprising as it sounds, `Adri´an
+P´erez-Salinas et al. (2019) <https://arxiv.org/abs/1907.02085>`_
+discuss this with their idea of ´data
+reuploading´. It is possible to load a single qubit with arbitrary
+dimensional data and then use it as a universal classifier.
+
+In this example, we will implement this idea with ``Pennylane`` - a
+python based tool for quantum machine learning, automatic
+differentiation, and optimization of hybrid quantum-classical
+computations.
+
+Circles
+~~~~~~~
+
+We consider a simple classification problem and will train a
+single-qubit variational quantum circuit to achieve this goal. The data
+is generated as a set of random points in a plane :math:`(x_1, x_2)` and
+labeled as 1 (blue) or 0 (red) depending on whether they lie inside or
+outside a circle. The goal is to train a quantum circuit to predict the
+label (red or blue) given an input point’s coordinate.
+
+.. figure:: ../../examples/figures/universal_circles.png
+   :scale: 65%
+   :alt: circles
+
+
+Transforming quantum states using unitary operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A single-qubit quantum state is characterized by a two-dimensional state
+vector and can be visualized as a point in the so-called Bloch sphere.
+Instead of just being a 0 (up) or 1 (down), it can exist in a
+superposition with say 30% chance of being in the :math:`|0 \rangle` and
+70% chance of being in the :math:`|1 \rangle` state. This is represented
+by a state vector :math:`\psi \rangle = 0.3|0 \rangle + 0.7|1 \rangle ` - the probability
+“amplitude” of the quantum state. In general we can take a vector
+:math:`(\alpha, \beta)` to represent the probabilities of a qubit being
+in a particular state and visualize it on the Bloch sphere as an arrow.
+
+.. figure:: ../../examples/figures/universal_bloch.png
+   :scale: 65%
+   :alt: bloch
+
+Data loading using unitaries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to load data onto a single qubit, we use a unitary operation U
+(:math:`x_1`, :math:`x_2`, :math:`x_3`) which is just a parameterized
+matrix multiplication representing the rotation of the state vector in
+the Bloch sphere. Eg., to load :math:`(x_1, x_2)` into the qubit, we
+just start from some initial state vector, :math:`|0 \rangle `,
```

Comment:
No spaces are allowed before the backtick in math mode, otherwise it will cause parts of the paragraph not to render
```suggestion
just start from some initial state vector, :math:`|0 \rangle`,
```